packages/web-lib/src/persistence/web-store.types.ts

```diff
@@ -0,0 +1,11 @@
+export interface IStorageStrategy {
+  getOneByKey<TStoredData>(key: string): Promise<OperationResult<TStoredData>>;
+  setOneByKey(key: string, value: unknown): Promise<OperationResult<void>>;
+  remove(key: string): Promise<OperationResult<void>>;
+  init(): Promise<void>;
+  storageAvailable(): Promise<boolean>;
```

Comment:
storageAvailable is an interesting api. good idea, not too sure about it's implementation tho.